comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go

```diff
@@ -268,6 +270,46 @@ func (c *collector) GetTargetCatalog() workloadmeta.AgentType {
 	return c.catalog
 }
 
+func collectKubeCapabilities(ctx context.Context, apiserverClient *apiserver.APIClient, wlmetaStore workloadmeta.Component) {
+	featureGates, err := common.ClusterFeatureGates(ctx, apiserverClient.Cl.Discovery(), 5*time.Second)
+	if err != nil {
+		log.Warnf("failed to collect cluster feature gates: %v", err)
+		featureGates = make(map[string]common.FeatureGate)
+	}
+
+	wlmFeatureGates := make(map[string]workloadmeta.FeatureGate)
+	for name, featureGate := range featureGates {
+		wlmFeatureGates[name] = workloadmeta.FeatureGate{
+			Name:    featureGate.Name,
+			Stage:   workloadmeta.FeatureGateStage(featureGate.Stage),
+			Enabled: featureGate.Enabled,
+		}
+	}
+
+	versionInfo, err := common.KubeServerVersion(apiserverClient.Cl.Discovery(), 5*time.Second)
+	if err != nil {
+		log.Warnf("failed to collect cluster version: %v", err)
```

Comment:
what is the value of `versionInfo` here in case of error ? we don't do a early return here so we'll pass the value to the WLM to notify subscribers, are we sure it is safe ?